### PR TITLE
feat: implement foreground tracking with haptic feedback

### PR DESCRIPTION
**Pull request by zeckkimon**
* Used the flag 'isTrackingEnabled' to enable/disable foreground tracking
* Tracking stops on device lock or when the app is in the background, then resumes when user opens the app

### CODE REVIEW

**Comment by zeckkimon** [comment]
Task linked: [CU-86b0qzpgm Foreground device monitoring](https://app.clickup.com/t/86b0qzpgm)

File: src/app/(tabs)/index.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { useUser } from "@state/useUser";
 import { Box } from "@gluestack-ui/themed";
 import { Redirect } from "expo-router";
 import { useBackgroundTasks } from "@src/components/providers/BackgroundTasksProvider";
-
+import DeviceMotionView from "@src/components/ui/DeviceMotionView";
```
Comment (whamester) [inline src/app/(tabs)/index.tsx:8]: in the future add a break line, to separate the imports from the component content

File: src/components/ui/DeviceMotionView.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { Text, View } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { DeviceMotion } from 'expo-sensors';
+import { useBackgroundTasks } from "@src/components/providers/BackgroundTasksProvider";
+import * as Haptics from "expo-haptics";
+
+export default function DeviceMotionView() {
+    const [{beta, gamma, alpha }, setRotationData] = useState({beta: 0, gamma: 0, alpha: 0 });
```
Comment (whamester) [inline src/components/ui/DeviceMotionView.tsx]: if alpha is not being used, remove its declaration

File: src/components/ui/DeviceMotionView.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Heavy);
+            showNotification();
+        }
+    },[beta, gamma])
+
+    const isBadPosture = () => {
+        return  orientation == 0 && inBadPostureRange(beta) ||
+                orientation != 0 && inBadPostureRange(gamma)
+    }
+    const inBadPostureRange = (data: number) => {
```
Comment (whamester) [inline src/components/ui/DeviceMotionView.tsx:64]: Add a break line between these two functions

**Review by whamester — APPROVED** [review]

File: src/app/(tabs)/index.tsx (hunk 1 above)
Comment (zeckkimon) [reply src/app/(tabs)/index.tsx:8]: This is updated.

File: src/components/ui/DeviceMotionView.tsx (hunk 2 above)
Comment (zeckkimon) [reply src/components/ui/DeviceMotionView.tsx]: This is updated.

File: src/components/ui/DeviceMotionView.tsx (hunk 3 above)
Comment (zeckkimon) [reply src/components/ui/DeviceMotionView.tsx:64]: This is updated.

**Review by ReinhardtBotha — APPROVED** [review]
nice! works on my phone 👍

**Review by Victorportus — APPROVED** [review]

**Pull request merged**